cmd/internal/flowmode/cluster_builder.go

```diff
@@ -54,12 +55,7 @@ func buildClusterService(opts clusterOptions) (*cluster.Service, error) {
 	}
 
 	if config.AdvertiseAddress == "" {
-		// TODO(rfratto): allow advertise interfaces to be configurable.
-		addr, err := advertise.FirstAddress(advertise.DefaultInterfaces)
-		if err != nil {
-			return nil, fmt.Errorf("determining advertise address: %w", err)
-		}
-		config.AdvertiseAddress = fmt.Sprintf("%s:%d", addr.String(), listenPort)
```

Comment:
Couldn't it be simpler to just do this here?

```
		if err != nil {
			level.Warn(opts.Log).Log("msg", "could not find advertise address using default interface names, "+
				"falling back to localhost", "err", err)
			config.AdvertiseAddress = fmt.Sprintf("127.0.0.1:%d", listenPort)
		}
	}
```

Reply:
It wouldn't really matter when clustering is disabled either way, as the ckit endpoints are never registered so no communication can happen.

Reply:
Wouldn't this mean that every Windows user that doesn't enable clustering will get a warning at startup? I would prefer to avoid the warning. 

But yeah, I can inline this function too.

Reply:
For now yes. Since the DefaultInterface handling is a best-effort, Linux-only thing, I'd say we can do a better job on that end (maybe add `Ethernet1` or equivalent for Windows?)

Reply:
IMO if Windows we should not even pretend to start clustering at the moment.

Reply:
Why not? it should work on Windows AFAICT if you specify full IP address of one of your interfaces? I tested some simple configs on my private windows box.